Fixed formatting in async documentation

rpetrusha opened the pull request:
# Fixed formatting in async documentation

## Summary
This PR fixes a number of migration and formatting issues in the async documentation:
- A number of template member names had a space between the constant part of the name (which appeared as bold text) and the variable part (which appeared as italicized text). I've removed the space.
- A number of template names had italicized text followed by fenced text (e.g., \*memberName\*\`Async\`). Because of the formatting of fenced text, this seemed to me to be confusing, so I replaced the fenced text with bolded text (e.g., \*memberName\*\*\*Async\*\*) to improve readability.
- I removed a duplicate entry from the TOC, and promoted a child topic by one level.

Fixes #3921


## Suggested Reviewers

@mairaw

Commit:
Removed embedded space

Commit:
Corrected formatting in async documentation

Commit:
Merge branch 'apm-formatting' of https://github.com/rpetrusha/docs into apm-formatting

Fixing merge conflict.

mairaw commented on docs/standard/asynchronous-programming-patterns/best-practices-for-implementing-the-event-based-asynchronous-pattern.md:
```diff
@@ -96,7 +96,7 @@ private void Form1_MethodNameCompleted(object sender, MethodNameCompletedEventAr
 -   Support progress reporting, if possible. This enables developers to provide a better application user experience when they use your class.  
   
--   If you implement a `ProgressChanged`/*MethodName*`ProgressChanged` event, ensure that there are no such events raised for a particular asynchronous operation after that operation's *MethodName*`Completed` event has been raised.  
+-   If you implement a `ProgressChanged`/*MethodName***ProgressChanged** event, ensure that there are no such events raised for a particular asynchronous operation after that operation's *MethodName***Completed** event has been raised.  
```
shouldn't this `ProgressChanged` be also changed?

mairaw commented on docs/standard/asynchronous-programming-patterns/best-practices-for-implementing-the-event-based-asynchronous-pattern.md:
```diff
@@ -64,27 +64,27 @@ private void Form1_MethodNameCompleted(object sender, MethodNameCompletedEventAr
 ### Simultaneously Executing Operations  
   
--   If your class supports multiple concurrent invocations, enable the developer to track each invocation separately by defining the *MethodName*`Async` overload that takes an object-valued state parameter, or task ID, called `userSuppliedState`. This parameter should always be the last parameter in the *MethodName*`Async` method's signature.  
+-   If your class supports multiple concurrent invocations, enable the developer to track each invocation separately by defining the *MethodName***Async** overload that takes an object-valued state parameter, or task ID, called `userSuppliedState`. It should always be the last parameter in the *MethodName***Async** method's signature.  
```
I think the it is not clear here what is referring to

mairaw reviewed (approved):
LGTM @rpetrusha. Let's see how this looks on the built content. My only fear is now loc translating those terms since we're now unlocking them.

Commit:
Addressed review comments

rpetrusha commented:
@mairaw, I've addressed your comments. I think this looks good now -- at least, I think that the intent is much clearer -- but localization is an issue. Do you want to take a look and decide whether you agree before merging?

mairaw commented:
It does look better with your changes. Let's merge it.
